analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -27,7 +27,7 @@ internal static partial class SyntaxNodeExtensions
         private static readonly ControlFlowGraphCache CfgCache = new();
 
         public static ControlFlowGraph CreateCfg(this SyntaxNode body, SemanticModel model, CancellationToken cancel) =>
-            CfgCache.FindOrCreate(body.Parent, model, cancel);
+            CfgCache.FindOrCreate(body.Parent ?? body, model, cancel);
```

Comment:
I'm not happy about this, because there's a generic hardening question:
>Do we need to start CFG creation at Body?

I think we don't and we should not. But that's not a topic for this PR.

Make this more explicit, because `??` can have unpredicted sideffects here. Use `body is CompilationUnitSyntax ? body : body.Parent`
And try to investigate the hardening question. In the best case scenario, the `body.Parent` will go away, `body` will be renamed to `node`.

It's not as easy as it sounds. It's called for different things like lambdas, from VB, and also will have implications on the old CFG. For that one, it will be fine to do something more hacky there.

Reply:
I did the update.
And then I'll do the investigation outside of the context of this PR.